### ISSUE TITLE
The `upper` option of `at-rule-name-case` under Less writes a name Less refuses, on every at-rule

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#578]
Less reads an at-rule name in lower case only. `less@4.9.1` refuses every one of these, each with `Missing '[...]' lookup in variable call`:

```less
@MEDIA (a) { b { c: d } }
@Media (a) { b { c: d } }
@IMPORT "x";
@SUPPORTS (a: b) { c { d: e } }
@FONT-FACE { a: b }
@KEYFRAMES x { from { a: b } }
@CHARSET "utf-8";
@LAYER l;
@PAGE { margin: 0 }
```

and `@PAGE :first { margin: 0 }` with `Unrecognised input`, while `@page :FIRST { margin: 0 }` compiles and is printed back as it stands. An upper-case letter in the name sends the parser down its variable-call path, which #357 noted in passing.

So `@stylistic/less/at-rule-name-case` under `upper` leaves a file Less refuses wherever it writes at all: `--fix` turns `@page :first { margin: 0 }` into `@PAGE :first { margin: 0 }`, and the second run reports nothing, the rule having got what it asked for. Measured on `177f15b` and on the branch of #394 alike through the harness runner and the compiler — that branch's sweep put every `--fix` output of the rule through `less.render`, and the three rows left whose input compiles and whose output does not are all this: the three control at-rules of the corpus under `upper`.

## What would end it

The option cannot be satisfied under Less, so it is a decision rather than a fix: either the namespace's `at-rule-name-case` refuses `upper` at `validateOptions`, or it reports under `upper` and declines the fix, the way the namespace answers wherever an option asks for something the syntax has nowhere to put. Either way the README of the namespace says so. A mixin call, a variable declaration and a call to a detached ruleset are already passed over by the guard and are not what this is about.
